test/test-lex.cpp

```diff
@@ -113,6 +113,22 @@ TEST_F(test_lex, lex_block_comments) {
   }
 }
 
+TEST_F(test_lex, lex_unopened_block_comment) {
+  {
+    error_collector v;
+    padded_string input(u8"hello */"_sv);
```

Comment:
**Must fix**: The following code is legal (though pretty silly):
```javascript
3*/hello/;
// equivalent to:
3 * /hello/;  // multiply number '3' with regexp literal '/hello/'
```
I think your patch produces a false positive for this example.

Reply:
I've come up with a workaround that eliminates false positives for regexp literals. However, in some cases, it emits duplicate warnings. It also redundantly parses the regexp. A cleaner solution would need more changes to the expression and regexp parsing code. I'm not sure what the best approach would be to handle those cases.

Reply:
Can you show the patches for your new approach (workaround)?